hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/MorphingPortableReader.java

```diff
@@ -198,10 +198,22 @@ public short readShort(String fieldName) throws IOException {
     }
 
     @Override
+    public boolean[] readBooleanArray(String fieldName) throws IOException {
+        FieldDefinition fd = cd.getField(fieldName);
+        if (fd == null) {
+            return null;
+        }
+        if (fd.getType() != FieldType.BOOLEAN_ARRAY) {
+            throw new IncompatibleClassChangeError();
```

Comment:
For a future PR it would be a good thing to tell exactly what was incompatible. Currently you only signal that there is an issue, but not which field and the type conflict. Putting in a simple message can really help out the end user because now he has the complex task to figure out what exactly went wrong.